cwl/src/main/scala/cwl/package.scala

```diff
@@ -49,9 +49,16 @@ package object cwl extends TypeAliases {
     */
 
   type StringOrExpression = Expression :+: String :+: CNil
-
   type Expression = ECMAScriptExpression :+: ECMAScriptFunction :+: CNil
 
+  object StringOrExpression {
+    object String {
+      def unapply(soe: StringOrExpression): Option[String] = soe.select[String]
+    }
+    object Expression {
```

Comment:
can we take this one further and offer `ScriptExpression` and `ScriptFunction` that match to the nested `ECMAScriptExpression` and `ECMAScriptFunction`, respectively?

After all my talk of pattern matching being useful in nested coproducts I didn't even do it!

Reply:
I hadn't done initially since it wasn't being used (😄), but I don't see why not. Added.